src/sparse/impl/KokkosSparse_spgemm_impl_speed.hpp

```diff
@@ -314,6 +313,21 @@ struct KokkosSPGEMM
           shmem_key_size = shmem_key_size + ((shmem_key_size - shmem_hash_size) * sizeof(nnz_lno_t)) / (sizeof (nnz_lno_t) * 2 + sizeof(scalar_t));
```

Comment:
Not technically part of the PR, but I had trouble following line 313. A comment on what is being accomplished would help newbies likes.

Reply:
This was here previously, it's a bit funky but acts as part of the heuristic that ultimately determines shmem sizes for:

```
used_hash_sizes : sizeof (nnz_lno_t) * 2 
begins (hash_sizes) : sizeof(nnz_lno_t) * shmem_hash_size
nexts : sizeof(nnz_lno_t) * shmem_key_size
keys : sizeof(nnz_lno_t) * shmem_key_size
vals : sizeof(scalar_t) * remainder
```